### PR TITLE
Add webhook and multi-thread support (on webhook mode)

### PR DESCRIPTION
**Pull request by yangm97**
Thanks to the power of openresty and some nice non-blocking libraries, GroupButler can be 100% non-blocking and load balanced from CPU cores to multiple servers. This offers great speed and stability improvements, along with reduced unexpected downtimes.

It also allows people building on top of GroupButler to “fool” GB with fake requisitions for debugging or route telegram requests as they come along. For instance, you can manually send fake updates using curl; have a “router” that allows choosing between beta and stable versions (from the same bot @username). The possibilities are endless!

Overview:
- Built on top of the last PR (Fix most luacheck warnings and…). Merging this will also merge that.

TODO:

- [x] Hunt down blocking code and libraries and replace them with non blocking alternatives.
- [x] Add token detection to incoming web hook to improve security 
- [x] Hide token from logs

Postponed:
- [x] Error logging #220 
- [x] Documentation

### CODE REVIEW

**Comment by Synk0** [comment]
Any news?

**Comment by yangm97** [comment]
@Synk0 waiting for #218 and #224 to be reviewed and merged

**Comment by TiagoDanin** [comment]
News?

**Comment by yangm97** [comment]
@TiagoDanin I found a bug with rules (which shouldn’t be hard to fix), but I’m waiting for a review. Maybe I should’ve dropped the WIP label 🤔

File: lua/groupbutler/main.lua [hunk 1]
```diff
@@ -99,19 +98,17 @@ local function on_msg_receive(msg, callback) -- The fn run whenever a message is
...
 					end
 
 					--if not check_callback(msg, callback) then goto searchaction end
 					local success, result = xpcall(plugin[callback], debug.traceback, msg, blocks) --execute the main function of the plugin triggered
 
 					if not success then --if a bug happens
-						print(result)
 						if config.bot_settings.notify_bug then
-							api.sendReply(msg, _("🐞 Sorry, a *bug* occurred"), true)
+							u.sendReply(msg, _("🐞 Sorry, a *bug* occurred"), true)
```
Comment (TiagoDanin) [inline lua/groupbutler/main.lua]: u.sendReply(msg, **_(**"🐞 Sorry, a *bug* occurred"), true)
Replace `_(` ~> `i18n`

**Comment by yangm97** [comment]
Blocked by #220 since I removed almost (all?) log from this branch. Ooops.

File: polling.lua [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
-local plugins = require 'plugins'
-local main = require 'main'
+local plugins = require "groupbutler.plugins"
+local main = require "groupbutler.main"
+local config = require "groupbutler.config"
+local u = require "groupbutler.utilities"
+local api = require "telegram-bot-api.methods".init(config.telegram.token)
 
-bot = api.getMe().result
+u.assert_startup()
```
Comment (TiagoDanin) [inline polling.lua]: u.assert_startup(1, true, false)

**Pull request merged**